src/Native/System.Native/pal_networking.cpp

```diff
@@ -1197,8 +1201,12 @@ extern "C" Error SystemNative_GetIPv4MulticastOption(int32_t socket, int32_t mul
     }
 
     *option = {.MulticastAddress = opt.imr_multiaddr.s_addr,
+#if HAVE_IP_MREQN
                .LocalAddress = opt.imr_address.s_addr,
                .InterfaceIndex = opt.imr_ifindex};
+#else
+               .LocalAddress = opt.imr_interface.s_addr};
```

Comment:
@stephentoub, the `struct ip_addr` (the type for `imr_interface`) has only one member http://ftp.netbsd.org/pub/NetBSD/NetBSD-current/src/sys/netinet/in.h. Maybe there is some sysctl call to get the interface index. I am unaware of it. @krytarowski, do you have any clue?